portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -70,11 +73,14 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     storeComments(name, email, comment, timestamp);
     response.sendRedirect("/index.html");
   }
-
-  private static String buildComment(Entity entity) {
+  
+  private static String buildComment(Entity entity, String prevLanguageCode, String languageCode) {
     String name = (String) entity.getProperty(ENTITY_NAME_HEADER);
     String email = (String) entity.getProperty(ENTITY_EMAIL_HEADER);
     String comment = (String) entity.getProperty(ENTITY_COMMENT_HEADER);
+    if (!prevLanguageCode.equals(languageCode)) {
+      comment = translateComment(comment, languageCode);
+    }
     long timestamp = (long) entity.getProperty(ENTITY_TIMESTAMP_HEADER);
```

Comment:
You may want to check if the language code is valid here and throw an exception if it is not valid just to be safe. line 110: Additionally, possibly check what the API returns if the language code is not valid.

Reply:
Done